### PR TITLE
[ci skip] Add `make fmt` at the top of the istio.io/istio repo.

### PR DESCRIPTION
**Pull request by guptasu**
**What this PR does / why we need it**:
In the new mono repo, we need a single `make fmt` command that would format the code under the entire mono repo. 

Currently the included dirs list is : "security pilot mixer broker"
Let me know if more sub directories needs to be added.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: bin/fmt.sh [hunk 1]
```diff
@@ -9,8 +9,8 @@ source $SCRIPTPATH/use_bazel_go.sh
 ROOTDIR=$SCRIPTPATH/..
 cd $ROOTDIR
 
-PKGS="tests/e2e"
-GO_FILES=$(find ${PKGS} -type f -name '*.go')
+PKGS="tests/e2e mixer pilot security"
```
Comment (mandarjog) [inline bin/fmt.sh]: add broker

File: mixer/bin/fmt.sh [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-GO_FILES=$(find adapter cmd example tools/codegen test pkg template -type f -name '*.go')
-
-UX=$(uname)
-
-#remove blank lines so gofmt / goimports can do their job
-for fl in ${GO_FILES}; do
-	if [[ ${UX} == "Darwin" ]];then
-		sed -i '' -e "/^import[[:space:]]*(/,/)/{ /^\s*$/d;}" $fl
-	else
-		sed -i -e "/^import[[:space:]]*(/,/)/{ /^\s*$/d;}" $fl
```
Comment (rshriram) [inline mixer/bin/fmt.sh]: Is this Darwin hack present in the main fmt file ?

**Comment by mandarjog** [comment]
@guptasu thanks for taking this!, now we can turn fmt validation back on.

File: bin/fmt.sh (hunk 1 above)
Comment (guptasu) [reply bin/fmt.sh]: D

File: mixer/bin/fmt.sh (hunk 2 above)
Comment (guptasu) [reply mixer/bin/fmt.sh]: YES main fmt looks like
```
#remove blank lines so gofmt / goimports can do their job
for fl in ${GO_FILES}; do
  if [[ ${UX} == "Darwin" ]];then
    sed -i '' -e "/^import[[:space:]]*(/,/)/{ /^\s*$/d;}" $fl
  else
    sed -i -e "/^import[[:space:]]*(/,/)/{ /^\s*$/d;}" $fl
fi
```

**Comment by ldemailly** [comment]
add tests/ ?

**Comment by guptasu** [comment]
Hi @ldemailly , I am not sure I follow your suggesting. Are you suggesting as part of `make fmt` we also run tests ?

**Comment by ldemailly** [comment]
no I'm suggesting make fmt formats all the .go files, including the ones in tests/
(or rather _all_ so not even listing any directories 1 by 1 as we will add more directories with more shared code, tools etc)

**Comment by guptasu** [comment]
+1 @mjog any concern?

**Comment by mandarjog** [comment]
Yes we should format (and check that things are formatted) all code that we have created.

We should not format generated code. This is missing from this PR. We don’t want to touch the files that are listed in ‘generated_files’

File: Makefile [hunk 3]
```diff
@@ -44,6 +44,9 @@ check:
 build: setup
 	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) //...
 
+fmt:
+	@bin/fmt.sh
```
Comment (ldemailly) [inline Makefile]: side note I think it's easier to debug without @ in front of commands so one knows what exact command is ran (I just changed a bunch of them, it's more important when there are variables/arguments so this one is maybe ok as is)

**Comment by ldemailly** [comment]
setting do-not-merge so it doesn't auto go in as I think there is a follow up commit coming

**Comment by guptasu** [comment]
@ldemailly PTAL

**Pull request merged**